File: r/bob/bob.R
bob <- function(input) {

  if (input==toupper(input) & grepl("[[:upper:]]+", input)) {
    return ("Whoa, chill out!")
  }
  else if (grepl("[^A-Z]+.*\\?\\s*$", input)) {
    return ("Sure.")
  }
  else if (!grepl("[A-Za-z0-9]", input)) {
    return ("Fine. Be that way!")
  }
  else {
    return ("Whatever.")
  }
  
}

File: python/bob/bob.py
import re

def hey(what):
  
  if re.search(r"[A-Z]+", what) and what == what.upper():
    response = 'Whoa, chill out!'
  elif re.search(r"\?\s*$", what):
    response = 'Sure.'
  elif not re.search(r"[A-Za-z0-9]", what):
    response = 'Fine. Be that way!'
  else:
    response = 'Whatever.'

  return response
